flink-core/src/main/java/org/apache/flink/api/common/typeutils/base/BigDecSerializer.java

```diff
@@ -69,17 +69,17 @@ public void serialize(BigDecimal record, DataOutputView target) throws IOExcepti
 		}
 		// fast paths for 0, 1, 10
 		// only reference equality is checked because equals would be too expensive
```

Comment:
did you see this comment?

Reply:
No.
In terms of performance, this is OK, but from the point of view of language specification, may be it's better to use `equals` method.

Reply:
The code adheres to the API provided by `BigInteger`, so this is fine as far as I'm concerned.
It would be a different story if, say, we would use reference equality for `2`, since that relies an implementation detail that isn't exposed anywhere.

Reply:
right, the second section code was not exposed anywhere, and just modify the first section code now